### DISCUSSION TITLE
Clear mouse selection rectangle

### DISCUSSION THREAD

**Discussion by denis-yasnitsky** [xyflow/xyflow#1320] [Q&A] [answered]
Hi! Is there any way to programmatically clear a selection rectangle (after user has selected something with mouse)?

**Comment by moklick** [comment]
Hey,

could you try this:

```js
const setSelectedElements = useStoreActions(a => a.setSelectedElements);

... 

setSelectedElements([]);
```

**Reply by denis-yasnitsky** [reply]
Probably a dumb mistake on my side, but it seems like you cannot use this hook within a component that uses ReactFlow even if it's wrapped in ReactFlowProvider. (i.e. in your examples setSelectedElements is called from a different component)

**Comment by ajberasategui** [comment]
Hey @moklick, thank you for the awesome library. 

I have kind of the same question. Using ```setSelectedElements([]);``` clears the selection but does not clear the blue selection rectangle in the canvas. Is there a way to completely remove the rectangle/box?

A quick workaround that I've found is to hide the rectangle using CSS so it's never shown after mouse release.

Than you very much!

**Comment by MatthieuJnon** [comment, marked as answer]
For anyone having this issue (@ajberasategui) You have to get both those actions
```
    const setSelectedElements = useStoreActions(actions => actions.setSelectedElements);
    const unsetUserSelection = useStoreActions(actions => actions.unsetUserSelection);
```
And then do
```
        setSelectedElements([]);
        unsetUserSelection();
```

[As this seems like the only way to turn `nodesSelectionActive` to `false` manually.](https://github.com/wbkd/react-flow/blob/main/src/store/reducer.ts#L225)

**Reply by pankaj-avhad** [reply]
unsetUserSelection is no more part of useStore if I am not wrong resetSelectedElements does the half of the job meaning does not remove the blue selection box so @moklick do you have plan to expose UserSelection unset/reset

**Reply by moklick** [reply]
nodes now have a `selected` flag. Set all nodes selected: false, and the user selection should be gone.

**Reply by pankaj-avhad** [reply]
hey that works, thanks you so much @moklick
